pandas/core/groupby.py

```diff
@@ -642,9 +642,20 @@ def apply(self, func, *args, **kwargs):
 
         func = self._is_builtin_func(func)
 
-        @wraps(func)
-        def f(g):
-            return func(g, *args, **kwargs)
+        # this is needed so we don't try and wrap strings. If we could
+        # resolve functions to their callable functions prior, this
+        # wouldn't be needed
+        if args or kwargs:
+            if is_callable(func):
+
+                @wraps(func)
+                def f(g):
+                    return func(g, *args, **kwargs)
+            else:
+                raise ValueError('func must be a callable if args or '
+                                 'kwargs are supplied')
+        else:
+            f = func
```

Comment:
The wrapping here is probably OK.

I'm suggesting that we convert strings to functions _here_, rather than passing strings onto the next function. So the `func = self._is_builtin_func(func)` call converts `'median'` to the function `.median` etc (rather than just `sum`).

Or do we need to resolve which `median` function to convert into downstream?

And yup, will do in another PR. Overall some of this stuff feels pretty messy, but maybe that's just because I'm new to it.